workflows/ci.yml

```diff
@@ -0,0 +1,44 @@
+name: CI
+
+on:
+  workflow_call:
+    inputs:
+      test-grammar:
+        type: boolean
+        default: true
+      test-rust:
+        type: boolean
+        default: false
+
+jobs:
+  grammar:
+    name: Test grammar
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-14]
+    if: ${{inputs.test-node}}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Set up NodeJS
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{vars.NODE_VERSION}}
```

Comment:
Not usually fan of `vars` since the information is stored outside of the repository IIRC. I assume the reason was for convenience, since you only need to bump it once and it'll be bumped for all workflows?

I wonder though, is this action even needed? Could we rely on the default nodejs shipped with the runners? That way they'd automatically be bumped as the runner image version is bumped.

Reply:
Some parsers (e.g. [pkl](https://github.com/apple/tree-sitter-pkl)) don't work with some node versions.

Reply:
Which version does it need? And does it need to be that exact version, or is it just a minimum version?

Reply:
`tree-sitter generate` produces an incompatible parser with Node versions newer than 18.

Reply:
dafuq. Broken software lol.

OK, then this approach does seem to make most sense then 👍